MyThreadPool/MyThreadPool/MyTask.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Diagnostics;
+
+namespace MyThreadPool;
+
+/// <summary>
+/// A class for creating tasks
+/// </summary>
+public class MyTask<TResult> : IMyTask<TResult>
+{
+    private readonly Func<TResult>? suppiler;
+    private volatile bool isCompleted = false;
+    private TResult? result;
+    private readonly Queue<Action> queueWithContinueWithTasks;
+    private readonly MyThread[] arrayThreads;
+    private readonly Queue<Action> queueWithTasks;
+    private readonly Object locker;
+    private CancellationTokenSource token;
+
+    /// <summary>
+    /// Constructor for creating a task
+    /// </summary>
+    public MyTask(Func<TResult> suppiler, MyThread[] arrayThreads, Queue<Action> queueWithTasks, Object locker, CancellationTokenSource token)
+    {
+        this.suppiler = suppiler;
+        queueWithContinueWithTasks = new();
+        this.arrayThreads = arrayThreads;
+        this.queueWithTasks = queueWithTasks;
+        this.locker = locker;
+        this.token = token;
+    }
+
+    public TResult? Result
+    {
+        get
+        {
+            while (!isCompleted && !token.IsCancellationRequested) {}
+            return result;
+        }
+    }
+    public bool IsCompleted
+    {
+        get
+        {
+            return isCompleted;
+        }
+    }
```

Comment:
```suggestion
    public bool IsCompleted { get; private set; }
```
7 строк превратились в одну, красиво же :)